src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -160,6 +162,40 @@ internal set
 
         FaultTolerantAmqpObject<RequestResponseAmqpLink> RequestResponseLinkManager { get; }
 
+        private async Task<Message> ProcessMessage(Message message)
+        {
+            if (this.RegisteredPlugins == null)
+            {
+                return message;
+            }
+
+            var processedMessage = message;
+            foreach (var plugin in this.RegisteredPlugins)
+            {
+                try
+                {
+                    processedMessage = await plugin.AfterMessageReceive(message).ConfigureAwait(false);
+                }
+                catch (Exception ex) when (ex is ServiceBusPluginException pluginException && pluginException.ShouldCompleteOperation)
+                {
+                    continue;
+                }
+            }
+            return processedMessage;
+        }
+
+        private async Task<IList<Message>> ProcessMessages(IList<Message> messageList)
+        {
```

Comment:
return right here if there are no registered plugins. need to iterate through all the messages.